File: lib/models/restaurant.dart
import 'package:food_delivery/models/dish.dart';

class Restaurant {
  final String id;
  final String name;
  final String coverImage;
  final double rating;
  final int reviewCount;
  final int deliveryTime;
  final List<String> tags;
  final bool freeDelivery;
  final bool promoted;
  final double distance;
  final int priceLevel;
  final String address;
  final String description;

  const Restaurant({
    required this.id,
    required this.name,
    required this.coverImage,
    required this.rating,
    required this.reviewCount,
    required this.deliveryTime,
    required this.tags,
    required this.freeDelivery,
    required this.promoted,
    required this.distance,
    required this.priceLevel,
    required this.address,
    required this.description,
  });
}

File: types/restaurant.ts
export interface Restaurant {
  id: string;
  name: string;
  coverImage: string;
  rating: number;
  reviewCount: number;
  deliveryTime: number;
  tags: string[];
  freeDelivery: boolean;
  promoted: boolean;
  distance: number;
  priceLevel: 1 | 2 | 3; // 1 = $, 2 = $$, 3 = $$$
  address: string;
  description: string;
}

export interface Category {
  id: string;
  name: string;
  image: string;
}

export interface Dish {
  id: string;
  restaurantId: string;
  name: string;
  description: string;
  price: number;
  image: string;
  popular: boolean;
  category: string;
  options?: DishOption[];
}

export interface DishOption {
  id: string;
  name: string;
  choices: {
    id: string;
    name: string;
    price: number;
  }[];
  required: boolean;
  multiple: boolean;
}

export interface CartItem {
  id: string;
  dishId: string;
  name: string;
  price: number;
  quantity: number;
  options?: {
    name: string;
    choice: string;
    price: number;
  }[];
}